### ISSUE TITLE
DataList with User-defined list and Dropdown List set to not allow empty doesn't display first available option

### ISSUE THREAD

**Issue by dannydrogt** [leekelleher/umbraco-contentment#363]
### Which Contentment version are you using?

4.5.0

### Which Umbraco version are you using? For example: 8.14.1 - don't just write v8

10.7.0

### Bug summary

Using a DataList with a User-defined List as Data source and a Dropdown List as List editor, set to **not** Allow empty doesn't select the first option as a default value, but instead the dropdown list stays empty.

### Steps to reproduce

Create a Document Type with a Contentment Data List Datatype.
Set the Data source to User-defined List, add some random items (I used "Item 1", "Item 2", "Item 3").
Set the List editor to Dropdown List.
Unselect the "Allow empty?" option.
Save everything and create a new content node based on this Document Type.

### Expected result / actual result

I expect the Dropdown List to display the first available option from the User-defined List.
The actual result is that the Dropdown List stays empty.

(Also tried this setup with Umbraco 10.6.1 with Contentment 4.4.5 - this gives the same result).

### Do you have Umbraco ModelsBuilder enabled?

- [X] Yes, it is enabled.

### What browsers are you seeing the problem on?

Microsoft Edge

**Comment by dannydrogt** [comment]
When I check the Angular model when creating new content the value property in the DOM is undefined. It looks like the check on line 33 in dropdown-list.js doesn't work. 

https://github.com/leekelleher/umbraco-contentment/blob/fd9fdb6aa0cbc0a425838bab43482fa04ff8c2ef/src/Umbraco.Community.Contentment/DataEditors/DropdownList/dropdown-list.js#L30

For the older 10.6.1 with 4.4.5 setup, I also noticed some initial Angular model values of an empty array (I have no idea why, there didn't seem to be anything strange going on). But for that particular scenario the check on line 30 would skip the default value completely, as Angular just sees an empty array.

**Comment by leekelleher** [comment]
Hi @dannydrogt, thanks for raising this!

Following through the logic in your comment (above, https://github.com/leekelleher/umbraco-contentment/issues/363#issuecomment-1782536267), you're absolutely correct, the value checks had flaws.

I've updated these in commit 5016594092875f1ba8b55130472efdfe6e49b6c5, so now that it checks if the array value has any items, otherwise falls back on the default value (empty string).  I've also separated out the `showEmpty` check from the `else if`, so it'll always be checked.

I'll get this out in the next patch release - no fixed release date yet, as I'll need to wrap up with any other Hacktoberfest contributions.
